### PR TITLE
Update release checklist

### PR DESCRIPTION
**Pull request by erak**
Adds some details that we identified as helpful during the 0.5.4 release. We could leave this open until the next release is done and adjust if needed.

### CODE REVIEW

File: ReleaseChecklist.md [hunk 1]
```diff
@@ -1,24 +1,52 @@
-Checklist for making a release:
+## Checklist for making a release:
 
+### Requirements
+ - [ ] AppVeyor account
+ - [ ] Lauchpad (Ubuntu One) account
+ - [ ] gnupg key (has to be version 1, gpg2 won't work) for `your-name@ethereum.org` created and uploaded
```
Comment (erak) [inline ReleaseChecklist.md:5]: @chriseth I'm not sure that this works in the end, because ``release_ppa.sh`` would need to be changed properly. And I can't tell right now, if changing the email and the key is enough. We could make these arguments to the script, though.

File: ReleaseChecklist.md [hunk 2]
```diff
@@ -1,24 +1,52 @@
...
  - [ ] Wait for the tests for the commit on ``release``, create a release in Github, creating the tag.
  - [ ] Wait for the CI runs on the tag itself (they should push artifacts onto the Github release page).
- - [ ] Run ``scripts/create_source_tarball.sh`` while being on the tag to create the source tarball.
- - [ ] Upload the source tarball (in the upload directory) to the release page.
+ - [ ] Run ``scripts/create_source_tarball.sh`` while being on the tag to create the source tarball. Make sure to create ``prerelease.txt`` before: (``echo -n > prerelease.txt``). This will create the tarball in a directory called ``upload``.
+ - [ ] Take the tarball from the upload directory (its name should be ``solidity_x.x.x.tar.gz``, otherwise ``prerelease.txt`` was missing in the step before) and upload the source tarball to the release page.
+ - [ ] Publish the release.
+
+### PPA
+ - [ ] Change ``scripts/release_ppa.sh`` to match your key's email and key id.
```
Comment (erak) [inline ReleaseChecklist.md:32]: They appear in several places. Maybe I should mention that. Also: I haven't fully tested it in the end.

File: ReleaseChecklist.md [hunk 3]
```diff
@@ -1,24 +1,52 @@
-Checklist for making a release:
+## Checklist for making a release:
 
+### Requirements
+ - [ ] AppVeyor account
```
Comment (chriseth) [inline ReleaseChecklist.md]: You can also just login through github. Is it actually needed?

File: ReleaseChecklist.md [hunk 4]
```diff
@@ -1,24 +1,52 @@
...
+
+### Pre-release
  - [ ] Ensure that a Github project exists for the release.
  - [ ] Check that all issues and pull requests from the Github project to be released are merged to ``develop``.
- - [ ] Create a commit in ``develop`` that updates the ``Changelog`` to include a release date (run ``./scripts/tests.sh`` to update the bug list). Sort the changelog entries alphabetically and correct any errors you notice.
+
+### Changelog
+ - [ ] Create a commit in ``develop`` that updates the ``Changelog`` to include a release date.
+ - [ ] Run ``./scripts/tests.sh`` to update the bug list.
+ - [ ] Sort the changelog entries alphabetically and correct any errors you notice.
```
Comment (chriseth) [inline ReleaseChecklist.md:14]: Should do that before actually creating the commit :)

Comment (chriseth) [reply ReleaseChecklist.md:14]: So it should be: sort, script, commit, pull request

File: ReleaseChecklist.md (hunk 3 above)
Comment (erak) [reply ReleaseChecklist.md]: That's true. But you'd need access to the project. But I think everyone has that. I can remove this requirement.

File: ReleaseChecklist.md (hunk 4 above)
Comment (chriseth) [reply ReleaseChecklist.md:14]: Also, perhaps don't mention "in `develop`" explicitly, sounds like you directly push to the develop branch.

Comment (erak) [reply ReleaseChecklist.md:14]: YES :)

File: ReleaseChecklist.md [hunk 5]
```diff
@@ -1,24 +1,52 @@
...
  - [ ] Create a pull request and wait for the tests, merge it.
+
+### Release page
+ - [ ] Create Github release page, select the ``release`` branch as new target and set tag to the new version (e.g. `v0.5.4`) (make sure you only `SAVE DRAFT` instead of `PUBLISH RELEASE` before the actual release)
```
Comment (chriseth) [inline ReleaseChecklist.md]: Could include a link to https://github.com/ethereum/solidity/releases/new

File: ReleaseChecklist.md [hunk 6]
```diff
@@ -1,24 +1,52 @@
...
  - [ ] Thank voluntary contributors in the Github release page (use ``git shortlog -s -n -e origin/release..origin/develop``).
  - [ ] Create a pull request from ``develop`` to ``release``, wait for the tests, then merge it.
```
Comment (chriseth) [inline ReleaseChecklist.md:23]: This is not related to the section title anymore.

File: ReleaseChecklist.md (hunk 5 above)
Comment (erak) [reply ReleaseChecklist.md]: Ah, forgot it. Thanks.

File: ReleaseChecklist.md (hunk 6 above)
Comment (chriseth) [reply ReleaseChecklist.md:23]: Ok, perhaps just rename the title to: `Create the Release` or something.

File: ReleaseChecklist.md [hunk 7]
```diff
@@ -1,24 +1,52 @@
...
  - [ ] Make a final check that there are no platform-dependency issues in the ``solidity-test-bytecode`` repository.
  - [ ] Wait for the tests for the commit on ``release``, create a release in Github, creating the tag.
  - [ ] Wait for the CI runs on the tag itself (they should push artifacts onto the Github release page).
```
Comment (chriseth) [inline ReleaseChecklist.md]:
```suggestion
 - [ ] Wait for the CI runs on the tag itself (travis and appveyor should push artifacts onto the Github release page).
```

File: ReleaseChecklist.md [hunk 8]
```diff
@@ -1,24 +1,52 @@
...
- - [ ] Run ``scripts/create_source_tarball.sh`` while being on the tag to create the source tarball.
- - [ ] Upload the source tarball (in the upload directory) to the release page.
+ - [ ] Run ``scripts/create_source_tarball.sh`` while being on the tag to create the source tarball. Make sure to create ``prerelease.txt`` before: (``echo -n > prerelease.txt``). This will create the tarball in a directory called ``upload``.
+ - [ ] Take the tarball from the upload directory (its name should be ``solidity_x.x.x.tar.gz``, otherwise ``prerelease.txt`` was missing in the step before) and upload the source tarball to the release page.
+ - [ ] Publish the release.
```
Comment (chriseth) [inline ReleaseChecklist.md]: "click the publish button on the github release page"?

File: ReleaseChecklist.md [hunk 9]
```diff
@@ -1,24 +1,52 @@
...
+
+### PPA
+ - [ ] Change ``scripts/release_ppa.sh`` to match your key's email and key id.
  - [ ] Run ``scripts/release_ppa.sh release`` to create the PPA release (you need the relevant openssl key).
  - [ ] Wait for the ``~ethereum/ubuntu/ethereum-static`` PPA build to be finished and published for *all platforms*. SERIOUSLY: DO NOT PROCEED EARLIER!!! *After* the static builds are *published*, copy the static package to the ``~ethereum/ubuntu/ethereum`` PPA for the destination series ``Trusty`` and ``Xenial`` while selecting ``Copy existing binaries``.
  - [ ] Check that the Docker release was pushed to Docker Hub (this still seems to have problems, run ``./scripts/docker_deploy_manual.sh release``).
```
Comment (chriseth) [inline ReleaseChecklist.md]: Not related to the title anymore :)

File: ReleaseChecklist.md [hunk 10]
```diff
@@ -1,24 +1,52 @@
...
- - [ ] Update the homebrew realease in https://github.com/ethereum/homebrew-ethereum/blob/master/solidity.rb (version and hash)
+
+### Homebrew
+ - [ ] Get write access to https://github.com/ethereum/homebrew-ethereum
```
Comment (chriseth) [inline ReleaseChecklist.md]: Shouldn't we have that already (see above)?

File: ReleaseChecklist.md (hunk 9 above)
Comment (chriseth) [reply ReleaseChecklist.md]: Also, you have to call it as `docker_deploy_manual.sh v0.x.x`, otherwise it will not push - we could also change the script.

File: ReleaseChecklist.md (hunk 4 above)
Comment (erak) [reply ReleaseChecklist.md:14]: > Also, perhaps don't mention "in `develop`" explicitly, sounds like you directly push to the develop branch.

Sure. We had that discussion already.

File: ReleaseChecklist.md (hunk 9 above)
Comment (erak) [reply ReleaseChecklist.md]: I'd suggest to revisit all the release related scripts separately. I've created an issue for that: https://github.com/ethereum/solidity/issues/6138

Comment (erak) [reply ReleaseChecklist.md]: Changed the command to `docker_deploy_manual.sh v0.x.x`, though.

**Comment by erak** [comment]
Updated. Needs to be squashed.

File: ReleaseChecklist.md (hunk 3 above)
Comment (chriseth) [reply ReleaseChecklist.md]: I mean where do you actually need to access appveyor?

**Review by chriseth — CHANGES_REQUESTED** [review]
Great! Please squash!

**Review by chriseth — APPROVED** [review]
